### PR TITLE
Migrate to package:web

### PR DESCRIPTION
**Pull request by ditman**
This PR:

* Migrates the web implementation of the package to `package:web`. This allows this package to be used by apps that want to be compiled to WASM.
* Updates the example app a little bit
* Fixes the platform_channel test so it does what it seems to want to do :)

This is what the example web app looks like now:

Screenshot 2023-12-06 at 10 27 29 PM

Thanks for this package @codebird!

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 description: Dynamic App Icon, provides a way to change favicon for web, and android app icon programatically.
-version: 0.0.8
+version: 0.1.0
 homepage: https://github.com/codebird/flutter_app_icons
 
 environment:
-  sdk: '>=2.12.1 <3.0.0'
-  flutter: ">=2.5.0"
+  sdk: ^3.2.0
+  flutter: ^3.16.0
```
Comment (ditman) [inline pubspec.yaml:8]: Hm, the android version doesn't seem to like this flutter update...

```console
$ flutter run -d emu

Launching lib/main.dart on Android SDK built for x86 in debug mode...
package identifier or launch activity not found.
Please check /work/ditman/codebird-flutter_app_icons/example/android/app/src/main/AndroidManifest.xml for errors.
No application found for TargetPlatform.android_x86.
Is your project missing an android/app/src/main/AndroidManifest.xml?
Consider running "flutter create ." to create one.
```

Would you be willing to update this package to a more recent version of Flutter @codebird?

Comment (codebird) [reply pubspec.yaml:8]: Hello @ditman, 

Thanks for the PR.

Will check it out, and see if I can fix on android. 

Yes I surely would like to keep it working with the most recent version of flutter, and keep it going... Time is always the issue.

Comment (codebird) [reply pubspec.yaml:8]: The issue is with DONT_KILL_APP flag, android doesn't like it anymore. Going to merge your pull request as it enhances the package... Will try to figure something out for Android.

**Comment by ditman** [comment]
Thanks for the super fast merge! 🙏

**Pull request merged**